zephyr/src/supp_wrapper.c

```diff
@@ -0,0 +1,583 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/*
+ * WPA Supplicant / main() function for Zephyr OS
+ * Copyright (c) 2003-2013, Jouni Malinen <j@w1.fi>
+ *
+ * This software may be distributed under the terms of the BSD license.
+ * See README for more details.
+ */
+
+#include <logging/log.h>
+
+#include "supp_main.h"
+#include "includes.h"
+#include "common.h"
+#include "common/defs.h"
+#include "wpa_supplicant/config.h"
+#include "wpa_supplicant_i.h"
+#include "driver_i.h"
+#include "fst/fst.h"
+#include "p2p_supplicant.h"
+#include "wpa_supplicant_i.h"
+#include "wifi_mgmt.h"
+
+//#define MAX_SSID_LEN 32
+
+int cli_main (int, const char **);
+extern struct wpa_supplicant *wpa_s_0;
+
+int zephyr_supp_disable_network(int id)
+{
+    char cmd[512] = {'\0'};
```

Comment:
Does this buffer really need to be that large? A lot of this space appears to be unused at the moment. Note that this is allocated on a stack so will automatically affect any thread that'd make use of those commands.